test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,569 @@
+					
+				}
+			
+			} catch (java.lang.IllegalArgumentException e) {
+				/* stack agent is not supported by the trigger method */
+				if (agent.equals("stack")) {
+					AssertJUnit.assertTrue("Did not receive the expected exception IllegalArgumentException", true);
+				} 
+				else {
+					Assert.fail("Unexpected IllegalArgumentException : " + e.getMessage());
+
+				}
+			} catch (Exception e) {
+				Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+				e.printStackTrace();
+			}
+
+			index++;
+		}
+	}
+
+	/**
+	 * Function to test if the requested dumps are triggered to the specified file names
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerDumpToFile(OpenJ9DiagnosticsMXBean diagBean, String test) {
+		boolean found = false;
+		String[] dumpAgents = {"java", "heap", "snap", "system", "stack"};
+		String[] files = {"javacore.%Y%m%d.%H%M%S", "heapdump.%H%M%S", "Snap.%pid.trc", "core.%pid.dmp", "stack.%seq.txt"};
+		int index = 0;
+		String dir = "." + File.separator + test + "_dumps";
+
+		for (String agent : dumpAgents) {
+			try {
+				String filePath = dir + File.separator + files[index];
+
+				String fileName = diagBean.triggerDumpToFile(agent, filePath);
+				File dumpFile = new File(fileName);
+
+				if (!(agent.equals("stack"))) {
+					found = findFile(dir, dumpFile.getName());
+					AssertJUnit.assertTrue(fileName + " not found", found);
+					
+				} 
+			
+			} catch (java.lang.IllegalArgumentException e) {
+				/* stack agent is not supported by the trigger method */
+				if (agent.equals("stack")) {
+					AssertJUnit.assertTrue("Did not receive the expected exception IllegalArgumentException", true);
+				} 
+				else {
+					Assert.fail("Unexpected IllegalArgumentException : " + e.getMessage());
+				}
+			} catch (Exception e) {
+				Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+				e.printStackTrace();
+			}
+
+			index++;
+		}
+	}
+
+	/**
+	 * Function to test if a class heap dump is created
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerClassicHeapDump(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+
+		try {
+			String fileName = diagBean.triggerClassicHeapDump();
+			found = findFile(".", new File(fileName).getName());
+			AssertJUnit.assertTrue(fileName + " not found", found);
+		} catch (InvalidDumpOptionException e){
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e){
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+			e.printStackTrace();
+		}
+
+	}	
+
+
+	/**
+	 * Function to test if the dump options were reset to what it was at JVM initialization
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private void test_resetDumpOption(OpenJ9DiagnosticsMXBean diagBean) {
```

Comment:
May want to add 's' to end of method name, to match underlying call setDumpOption**s**